lib/src/layer/tile_layer/tile_model.dart

```diff
@@ -0,0 +1,39 @@
+import 'dart:math';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_map/flutter_map.dart';
+import 'package:flutter_map/src/layer/tile_layer/tile_painter.dart';
+
+/// Model for tiles displayed by [TileLayer] and [TilePainter]
+class TileModel {
+  /// Controls how this tile is replaced by another tile
+  ///
+  /// See also:
+  ///
+  ///  * [Widget.key]
+  ///  * The discussions at [Key] and [GlobalKey].
+  final ObjectKey key;
```

Comment:
I'm not convinced why this is needed? It's never evaluated at any point, and there's no equality methods in this class that this can influence.

(We should probably implement equality here as well.)

Reply:
Looks like `ObjectKey` can be removed and equality implemented.